dissect/target/helpers/fsutil.py

```diff
@@ -963,3 +965,23 @@ def resolve_link(
         entry = resolve_link(fs, entry, previous_links)
 
     return entry
+
+
+def decompress_and_read(path: TargetPath) -> bytes:
+    if path.suffix == ".gz":
+        return gzip.GzipFile(fileobj=path.open()).read()
+    elif path.suffix == ".bz2":
+        bytestream = path.open().read()
+        return bz2.decompress(bytestream)
+    else:
+        return path.open().read()
+
+
+def decompress_and_readlines(path: TargetPath) -> [str]:
+    if path.suffix == ".gz":
+        return gzip.GzipFile(fileobj=path.open()).read().decode().splitlines()
+    elif path.suffix == ".bz2":
+        bytestream = path.open().read()
+        return bz2.decompress(bytestream).decode().splitlines()
+    else:
+        return path.open("rt").readlines()
```

Comment:
```suggestion
def open_decompress(path: TargetPath, mode: str = "rb"):
    if path.suffix == ".gz":
        import gzip
        
        return gzip.open(path.open(), mode)
    elif path.suffix == ".bz2":
        import bz2
        
        return bz2.open(path.open(), mode)
    else:
        return path.open(mode)
```
What about something like this? That way it's only one function and it allows for some flexibility in how to use it, e.g. (imagine the string is a valid `TargetPath`:

```python
bytes_buf = open_decompress("/dir/file.gz").read()
for line in open_decompress("/dir/file.gz", "rt"):
    print(line)
```

Moving the imports inline is because these modules might not always be available (which is the case for our custom internal CPython).